integration/spark2/src/main/scala/org/apache/spark/sql/CarbonDatasourceHadoopRelation.scala

```diff
@@ -75,4 +75,16 @@ case class CarbonDatasourceHadoopRelation(
     ", " + "Table name :" + carbonTable.getFactTableName + ", Schema :" + tableSchema + " ]"
   }
 
+  override def insert(data: DataFrame, overwrite: Boolean): Unit = {
+    if (carbonRelation.output.size > CarbonCommonConstants.DEFAULT_MAX_NUMBER_OF_COLUMNS) {
+      sys.error("Maximum supported column by carbon is: " +
+          CarbonCommonConstants.DEFAULT_MAX_NUMBER_OF_COLUMNS)
+    }
+    if(data.logicalPlan.output.size >= carbonRelation.output.size) {
+      LoadTableByInsert(this, data.logicalPlan).run(sparkSession)
+    } else {
+      sys.error("Cannot insert into target table because column number are different")
```

Comment:
Better to give more detail information in the log to describe the schema difference